unlock-js/src/accounts.js

```diff
@@ -13,3 +13,18 @@ export default function createAccountAndPasswordEncryptKey(password) {
     passwordEncryptedPrivateKey,
   }
 }
+
+/**
+ * Given an encrypted private key and a password, decrypts the private key and
+ * gets the public address from it.
+ * @param {*} encryptedPrivateKey
+ * @param {string} password
+ * @throws Throws an error if password does not decrypt private key.
+ */
+export function getAddressFromPrivateKey(encryptedPrivateKey, password) {
+  const web3 = new Web3()
+
+  const { address } = web3.eth.accounts.decrypt(encryptedPrivateKey, password)
```

Comment:
Q: what happens if address isn't present for whatever reason? Should we throw something, or are we okay with leaving it undefined?

Reply:
I was curious about that too, and I verified that web3 will throw an error along the lines of `Key derivation failed - possibly wrong password`. I _believe_ that should cover it, barring bugs in web3. In the end, if account is `undefined` that should halt everything since if no account gets into the state they won't be able to do anything.

Would definitely like to hear the perspective of others on this as well.